### PR TITLE
Prepare HA deployment

### PR DESCRIPTION
**Pull request by mp911de**
## Summary

Add Spring Session integration using Redis to retain sessions while redeployments. Introduces a dependency to have a running Redis server, even for a local startup.

Added services `pivotal-cla-redis` and `pivotal-cla-mysql`.

Integrate blue/green deployment using the Cloud Foundry Gradle plugin using `variants`. The script is adopted from Sagan and checks for required properties. The deployment script first pushes the application and if the application start suceeds, it uses `cfSwapDeployed` to switch from blue to green and remap the route.

Adopted `.travis.yml` to:
1. run `gradle check` if build is not on the `master` branch and not a pull request build
2. run the deployment build for `master`
## Local deployment

Define variables in `gradle.properties`:

```
cfUsername=…
cfPassword=…
security.oauth2.main.clientId=…
security.oauth2.main.clientSecret=…
security.oauth2.pivotal-cla.tokenSecret=…
```

Deployment:
- Staging:  `./gradlew deploy -Pspace=contributor-license-agreements-staging`
- Production:  `./gradlew deploy -Pspace=contributor-license-agreements`
## Deployment using TravisCI

Deployment:
- Staging: if branch starts with `staging-` 
- Production: if branch is `master`

### CODE REVIEW

**Comment by mp911de** [comment]
Open issue: Document the deployment in readme

File: build.gradle [hunk 1]
```diff
@@ -14,7 +14,8 @@ buildscript {
 apply plugin: 'java'
 apply plugin: 'eclipse'
 apply plugin: 'spring-boot'
-apply from: 'gradle/cf.gradle'
+// apply from: 'gradle/cf.gradle'
```
Comment (mp911de) [inline build.gradle]: remove commented line

File: gradle/deploy.gradle [hunk 2]
```diff
@@ -0,0 +1,156 @@
+def checkTasks = []
```
Comment (mp911de) [inline gradle/deploy.gradle:1]: Replace space indents by tabs

**Comment by mp911de** [comment]
Issues fixed, added deployment documentation to the readme.

**Comment by rwinch** [comment]
Thanks for the PR! This is now merged into master

**Pull request closed without merging**